x-pack/plugins/lens/public/editor_frame_service/service.tsx

```diff
@@ -123,76 +121,58 @@ export class EditorFrameService {
 
   public start(core: CoreStart, plugins: EditorFrameStartPlugins): EditorFrameStart {
     const createInstance = async (): Promise<EditorFrameInstance> => {
-      let domElement: Element;
       const [resolvedDatasources, resolvedVisualizations] = await Promise.all([
         collectAsyncDefinitions(this.datasources),
         collectAsyncDefinitions(this.visualizations),
       ]);
 
-      const unmount = () => {
-        if (domElement) {
-          unmountComponentAtNode(domElement);
-        }
-      };
+      const firstDatasourceId = Object.keys(resolvedDatasources)[0];
+      const firstVisualizationId = Object.keys(resolvedVisualizations)[0];
+
+      const { EditorFrame, getActiveDatasourceIdFromDoc } = await import('../async_services');
+
+      const palettes = await plugins.charts.palettes.getPalettes();
 
       return {
-        mount: async (
-          element,
-          {
-            doc,
-            onError,
-            dateRange,
-            query,
-            filters,
-            savedQuery,
-            onChange,
-            showNoDataPopover,
-            initialContext,
-            searchSessionId,
-          }
-        ) => {
-          if (domElement !== element) {
-            unmount();
-          }
-          domElement = element;
-          const firstDatasourceId = Object.keys(resolvedDatasources)[0];
-          const firstVisualizationId = Object.keys(resolvedVisualizations)[0];
-
-          const { EditorFrame, getActiveDatasourceIdFromDoc } = await import('../async_services');
-
-          const palettes = await plugins.charts.palettes.getPalettes();
-
-          render(
-            <I18nProvider>
-              <EditorFrame
-                data-test-subj="lnsEditorFrame"
-                onError={onError}
-                datasourceMap={resolvedDatasources}
-                visualizationMap={resolvedVisualizations}
-                initialDatasourceId={getActiveDatasourceIdFromDoc(doc) || firstDatasourceId || null}
-                initialVisualizationId={
-                  (doc && doc.visualizationType) || firstVisualizationId || null
-                }
-                key={doc?.savedObjectId} // ensures rerendering when switching to another visualization inside of lens (eg global search)
-                core={core}
-                plugins={plugins}
-                ExpressionRenderer={plugins.expressions.ReactExpressionRenderer}
-                palettes={palettes}
-                doc={doc}
-                dateRange={dateRange}
-                query={query}
-                filters={filters}
-                savedQuery={savedQuery}
-                onChange={onChange}
-                showNoDataPopover={showNoDataPopover}
-                initialContext={initialContext}
-                searchSessionId={searchSessionId}
-              />
-            </I18nProvider>,
-            domElement
+        EditorFrameContainer: ({
```

Comment:
I am passing the component inside of the object, because the further state management changes will require to pass resolvedDatasources and resolvedVisualizations in this object, so we can then initialize the store in the mounter with these two.